samza-core/src/main/java/org/apache/samza/system/SystemAdmins.java

```diff
@@ -20,26 +20,30 @@
 package org.apache.samza.system;
 
 import java.util.Map;
+import java.util.Set;
 import org.apache.samza.SamzaException;
 import org.apache.samza.config.Config;
 import org.apache.samza.config.JavaSystemConfig;
+import org.apache.samza.config.MapConfig;
 
 
+/**
+ * Provides a mapping from system name to a {@link SystemAdmin}. Remember to start/stop this.
+ */
 public class SystemAdmins {
   private final Map<String, SystemAdmin> systemAdminMap;
 
-  public Map<String, SystemAdmin> getSystemAdminsMap() {
-    return systemAdminMap;
-  }
-
   public SystemAdmins(Config config) {
     JavaSystemConfig systemConfig = new JavaSystemConfig(config);
     this.systemAdminMap = systemConfig.getSystemAdmins();
   }
 
-  // Used only for test
-  public SystemAdmins(Map<String, SystemAdmin> systemAdminMap) {
-    this.systemAdminMap = systemAdminMap;
+  /**
+   * @return New instance of {@link SystemAdmins} with an empty admin mapping. Since it has no admins, it does not need
```

Comment:
Javadoc format: description, param, return.

Can remove comment about not needing start/stop, since they should work (no-op) even if called.

Reply:
I had the comment in there for a few cases where it is a bit difficult to actually add the start/stop. For example, there are a few cases where an empty instance is used as the default argument value for a Scala method, and I wanted to clarify that the Scala method does not need to actually manage the lifecycle of that SystemAdmins, even though it is creating a new instance of the SystemAdmins.
Do you think the clarification is unnecessary?

Reply:
> there are a few cases where an empty instance is used as the default argument value for a Scala method
Is it simple to remove the default value? I can't imagine why a method would have an empty SystemAdmins.

Reply:
The non-test use cases ignore this default value, since they all pass an actual SystemAdmins. However, there are a significant number of tests which depend on this default value (I'm guessing those particular tests just don't need the admins). I was hoping to avoid refactoring all of those tests right now. Also, we would also possibly want to reorder argument lists to put the now-required SystemAdmins argument before other optional arguments, which is another bit of refactoring.
My hope with this current change is to give some clarification to where "real" SystemAdmins are created and to prevent future usage of this "test constructor" which is a bit too flexible.

Reply:
Synced offline about the start/stop comment. Will remove it in order to reduce confusion. In general, users should do start/stop anyways, but for certain cases like defaults for testing, it can be ok to leave out the start/stop.
Resolved